### DISCUSSION TITLE
Suggestion: show Relationships in Dependencies view

### DISCUSSION THREAD

**Discussion by zenzeinet** [TabularEditor/TabularEditor3#1098] [Ideas]
Hi,
Small suggestion from something that was great in TE2: TE2's dependencies view, when looking at a table, shows the relationships that start/end to a table.
It was super-useful, and I'm missing it in TE3. 
Also, the Shift-F12 shortcut, I with it would still be F3 (o a single key without a modifier so that I don't need two hands to invoke it9
Thanks!

**Comment by otykier** [comment]
Thanks for the suggestion. We thought the diagram view was more useful for this than the dependency view, but I guess many users have been accustomed to viewing relationships through the dependency view :-)

Regarding shortcut, we still have full configurability on the backlog, but we've had to prioritize other items for now.

**Comment by zenzeinet** [comment]
Hi,

Yes, having the relationships in the dependency view is often faster/more agile than in the diagram view to quickly check what-
filters-what. 

What I'm doing meanwhile is filtering the TOM Explorer by the table name (which I have to manually enter) and see the resulting relationships (and I have to manually adjust the width of the TOM Explorer to wider to see all the information). 

Those are two manual steps (typing and widening, and then undoing those steps) that would be much more efficient with the dependencies with relationships view, specially as it can be undocked 

Apart from that, there are some improvements that would be great in the diagram view:

- Make the width of the table wider, or at least as-wide-as-necessary as to fit the table name automatically. It often is the case whe name is too long for the default width (in my models / screen resolution).
- When you click on a table, highlight all the relationships / related tables. This is currently implemented when clicking on the specific column, but would useful to be done at table level for all relationships. Also, put in bold the related tables, not only the links among the tables (easier to read)
- Option to hide the columns that don't have a relationship - not useful in relationship view for tables with many columns, or even to hide all columns (table-only view), which would make large models fit much more nicely in a single view.

Just a few ideas. But first, please, relationship in dependencies view please, as it was in TE2 ;)

**Comment by otykier** [comment]
Great suggestions all around! Thanks for taking the time to provide this feedback.

**Comment by zenzeinet** [comment]
My pleasure, after using TE2 for a long time (and now finally TE3 Ent).

**Comment by mlonsk** [comment]
I believe most of these features have been in Tabular Editor 3 for a while, sorry for not updating this idea when they came out.
